src/Components/WishlistProductCard/WishlistProductCard.jsx

```diff
@@ -0,0 +1,237 @@
+import React,{ useEffect, useState} from 'react'
+import { Link, useNavigate } from "react-router-dom"
+import axios from "axios"
+import jwt_decode from "jwt-decode";
+import './WishlistProductCard.css'
+import {
+    useToast,
+    useWishlist,
+    useCart
+} from '../../index'
+
+export default function WishlistProductCard({ productdetails }) 
+{
+    const navigate = useNavigate()
+
+    const { userWishlist, dispatchUserWishlist } = useWishlist()
+    const { dispatchUserCart } = useCart()
+    const { showToast } = useToast()
+    const {
+        _id, 
+        bookName,
+        author,
+        originalPrice,
+        discountedPrice,
+        discountPercent,
+        imgSrc, 
+        imgAlt,
+        badgeText, 
+        outOfStock
+    } = productdetails
+    const [wishlistHeartIcon, setWishlistHeartIcon] = useState("fa-heart-o")
+    const [wishlistBtn, setWishlistBtn]             = useState("add-to-wishlist-btn")
+
+    useEffect(()=>{
+        const index = userWishlist.findIndex(product=> {
```

Comment:
avoid nested if else  repalce it by  ==< switch case or ternary operator  or if block only required 

in else all had included all the case of failure , but be specific what kind of error you are getting and according to write logic